tools/proofers/codepoint_validator.js

```diff
@@ -58,9 +61,96 @@ $(function () {
     });
 
     $("#cc-remove").click(function () {
-        let textarea = $("#text_data");
-        textarea.val(textarea.val().replace(pattern , ""));
+        textArea.value = textArea.value.replace(badPattern , "");
         $("#checker").hide();
         $("#proofdiv").show();
     });
+
+    var above = {
+        "=": "\u0304", // macron
+        ":": "\u0308", // diaeresis
+        ".": "\u0307", // dot
+        "`": "\u0300", // grave
+        "'": "\u0301", // acute
+        "^": "\u0302", // circumflex
+        ")": "\u0306", // breve
+        "~": "\u0303", // tilde
+        ",": "\u0327", // cedilla
+        "v": "\u030C", // caron
+        "*": "\u030A", // ring
+        "(": "\u0311", // inverted breve
+    };
+
+    var below = {
+        "=": "\u0331", // macron
+        ":": "\u0324", // diaeresis
+        ".": "\u0323", // dot
+        "`": "\u0316", // grave
+        "'": "\u0317", // acute
+        "^": "\u032D", // circumflex
+        ")": "\u032E", // breve
+        "~": "\u0330", // tilde
+        ",": "\u0327", // cedilla
+        "v": "\u032C", // caron
+        "*": "\u0325", // ring
+        "(": "\u032F", // inverted breve
+    };
+
+    var ligatures = {
+        "ae": "\u00e6",
+        "AE": "\u00c6",
+        "oe": "\u0153",
+        "OE": "\u0152",
+    };
+
+    $(textArea).on("input", function() {
+        // if text changed and character before caret is ]
+        // (i.e. just inserted or pasted ] or deleted following character)
+        // check for markup like [..] for ligature or diacritical
+        // i0, i1, i3 point to these characters
+        var text = textArea.value;
+        var startPos = textArea.selectionStart;
+        var i0, i1, i3, uchar;
+
+        function substitute() {
```

Comment:
It's a bit unintuitive to have a function like this operate only "global" variables. Consider passing in the values you want the function to operate over:
```
function substitute(startIndex, replaceChar, startPos)
```

Reply:
Douglas Crockford says it is one of the 'very good' features of javascript that functions defined within another function have access to the outer function's local variables, but I agree with you that passing the parameters would make it more obvious how it is meant to work, although it could be less efficient, but probably no-one would notice the difference.

Reply:
IANAJavascriptDeveloper in any fashion, so if that's common practice that's fine with me.